frappe/core/doctype/communication/email.py

```diff
@@ -559,3 +572,44 @@ def mark_email_as_seen(name=None):
 		frappe.response["filename"] = "imaginary_pixel.png"
 		frappe.response["filecontent"] = buffered_obj.getvalue()
 
+def get_parent_doc(link_doctype, link_name):
+	"""Returns document of `link_doctype`, `link_doctype`"""
+	if link_doctype and link_name:
+		parent_doc = frappe.get_doc(link_doctype, link_name)
+	return parent_doc if parent_doc else None
+
+def get_contacts(email_strings):
+	email_addrs = []
+
+	for email_string in email_strings:
+		if email_string:
+			for email in email_string.split(","):
+					parsed_email = parseaddr(email)[1]
+					if parsed_email:
+						email_addrs.append(parsed_email)
+
+	contacts = []
+	for email in email_addrs:
+		contact_name = frappe.db.get_value('Contact', {'email_id': email})
+
+		if not contact_name:
+			contact = frappe.get_doc({
+					"doctype": "Contact",
+					"first_name": frappe.unscrub(email.split("@")[0]),
+					"email_id": email
+				}).insert(ignore_permissions=True)
+			contact_name = contact.name
+
+		contacts.append(contact_name)
+
+	return contacts
+
+def get_contact_links(communication, contact_name):
+	contact_links = frappe.get_list("Dynamic Link", filters={
```

Comment:
`get_contact_links` shouldn't add links to communication. Either rename it to `add_contact_links_to_communication` or do the linking where this is called.